### PR TITLE
Streaming api support

### PR DESCRIPTION
**Pull request by ryandoherty**
Adds streaming api support over wifi/usb for sample data.

### CODE REVIEW

File: autosportlabs/racecapture/databus/databus.py [hunk 1]
```diff
@@ -214,19 +216,45 @@ def start(self, data_bus, rc_api, streaming_supported):
...
+        Logger.debug("DataBusPump: Checking if new streaming api is supported")
+
+        def handle_capabilities(capabilities_dict):
+            self.rc_capabilities = Capabilities()
+            self.rc_capabilities.from_json_dict(capabilities_dict['capabilities'])
+
+            if self.rc_capabilities.has_streaming:
+                # Send streaming command
+                Logger.debug("DataBusPump: connected device supports streaming")
+                self._rc_api.start_telemetry(50)
```
Comment (brentpicasso) [inline autosportlabs/racecapture/databus/databus.py]: let's extract this number. suggest setting the number at the top, and then derive the 0.02 from this number (1/50) so the same number drives both rates. :+1:

File: autosportlabs/racecapture/databus/databus.py [hunk 2]
```diff
@@ -214,19 +216,45 @@ def start(self, data_bus, rc_api, streaming_supported):
...
+            else:
+                Logger.debug("DataBusPump: connected device does not support streaming api, starting polling")
+                t = Thread(target=self.sample_worker)
+                t.start()
+                self._sample_thread = t
+
+        def handle_capabilities_fail():
+            Logger.error("DataBusPump: Failed to get capabilities, can't determine if device supports streaming API")
```
Comment (brentpicasso) [inline autosportlabs/racecapture/databus/databus.py]: Seeing this made me think, maybe we can log an error through the crash handler, without blowing an exception up the stack?

Comment (brentpicasso) [inline autosportlabs/racecapture/databus/databus.py]: Not critical, just a thought.

**Comment by brentpicasso** [comment]
Found an issue where if I unplug / replug RCT or RCP the streaming is not re-activated.

Otherwise, this looks pretty solid so far!

File: autosportlabs/racecapture/databus/databus.py (hunk 1 above)
Comment (ryandoherty) [inline autosportlabs/racecapture/databus/databus.py]: 👍

File: autosportlabs/racecapture/databus/databus.py (hunk 2 above)
Comment (ryandoherty) [inline autosportlabs/racecapture/databus/databus.py]: Hmm yeah I was thinking about this too, it is technically an error. I will look into Sentry and see if we can log this to Sentry

**Comment by ryandoherty** [comment]
Made fixes per comments, in order to fix the reconnect issue I need some code from the autologging PR, will wait on that to be merged then fix this one.

**Comment by ryandoherty** [comment]
Ok @brentpicasso , reconnect logic added and tested over USB. having wifi issues with RCT, haven't tested yet.

**Comment by brentpicasso** [comment]
Tested, works, :shipit:

**Pull request merged**